api/v1beta1/machine_types.go

```diff
@@ -33,6 +33,9 @@ const (
 	// ExcludeNodeDrainingAnnotation annotation explicitly skips node draining if set.
 	ExcludeNodeDrainingAnnotation = "machine.cluster.x-k8s.io/exclude-node-draining"
 
+	// ExcludeWaitForNodeVolumeDetachingAnnotation annotation explicitly skips the waiting for node volume detaching if set.
+	ExcludeWaitForNodeVolumeDetachingAnnotation = "machine.cluster.x-k8s.io/exclude-wait-for-node-volume-detach"
```

Comment:
FWIW The annotation for draining was introduced as a way out before we had a timeout in the API. So normally I'd be against adding a new one the same functionality can in practice be achieved semantically via .VolumeDetachTimeout, however the main limitation we have today is that it can't be changed in a MachineDeployment without triggering a rolling upgrade and there fore defeating the purpose of the change so +1 for the annotation until we solve https://github.com/kubernetes-sigs/cluster-api/issues/5880